deps/src/main/java/com/microsoft/azure/sdk/iot/deps/transport/amqp/AmqpsConnection.java

```diff
@@ -130,6 +130,7 @@ public void setListener(AmqpListener listener)
 
     /**
      * Returns the status of the connection
+     * @throws Exception if any exception occurred during sasl negotiation
      * @return status of the connection
      */
     public boolean isConnected() throws Exception
```

Comment:
Why is this a generic Exception? Can you throw specific?

Reply:
Will do.

---
In reply to: [316295049](https://github.com/Azure/azure-iot-sdk-java/pull/540#discussion_r316295049) [](ancestors = 316295049)

Reply:
saslListener.getSavedException throws a generic Exception.   So that's why we have to do the same.

---
In reply to: [316295049](https://github.com/Azure/azure-iot-sdk-java/pull/540#discussion_r316295049) [](ancestors = 316295049)